x-pack/solutions/search/plugins/search_query_rules/server/lib/fetch_query_rules_ruleset.ts

```diff
@@ -11,9 +11,17 @@ import { ElasticsearchClient } from '@kbn/core/server';
 export const fetchQueryRulesRuleset = async (
   client: ElasticsearchClient,
   rulesetId: string
-): Promise<QueryRulesQueryRuleset> => {
-  const result = await client.queryRules.getRuleset({
-    ruleset_id: rulesetId,
-  });
-  return result;
+): Promise<QueryRulesQueryRuleset | null> => {
+  try {
+    const result = await client.queryRules.getRuleset({
+      ruleset_id: rulesetId,
+    });
+    return result;
+  } catch (error) {
+    if (error.meta?.statusCode === 404) {
```

Comment:
For the cases where we need this 404, I would revert this. 

Instead create a new lib function in a new file and use that to check.
`is_query_ruleset_exist`.


Something like:
```typescript
try {
  const ruleset = await client.queryRules.getRuleset({
      ruleset_id: rulesetId,
  });
  if ( result ) return true;
} catch (error) {
  if (error.meta?.statusCode === 404) return false;
  throw error;
}
```